tests/components/zwave/test_climate.py

```diff
@@ -166,25 +246,72 @@ def test_temperature_value_changed(device):
 
 def test_operation_value_changed(device):
     """Test values changed for climate device."""
-    assert device.hvac_mode == 'test1'
-    device.values.mode.data = 'test_updated'
+    assert device.hvac_mode == HVAC_MODE_HEAT
+    assert device.preset_mode == PRESET_NONE
+    device.values.mode.data = HVAC_MODE_COOL
     value_changed(device.values.mode)
-    assert device.hvac_mode == 'test_updated'
+    assert device.hvac_mode == HVAC_MODE_COOL
+    assert device.preset_mode == PRESET_NONE
+    device.values.mode.data = HVAC_MODE_OFF
+    value_changed(device.values.mode)
+    assert device.hvac_mode == HVAC_MODE_OFF
+    assert device.preset_mode == PRESET_NONE
+
+
+def test_operation_value_changed_preset(device_mapping):
+    """Test preset changed for climate device."""
+    device = device_mapping
+    assert device.hvac_mode == HVAC_MODE_HEAT
+    assert device.preset_mode == PRESET_NONE
+    device.values.mode.data = PRESET_ECO
+    value_changed(device.values.mode)
+    assert device.hvac_mode is None
+    assert device.preset_mode == PRESET_ECO
+    device.values.mode = None
+    assert device.preset_mode == PRESET_NONE
 
 
 def test_operation_value_changed_mapping(device_mapping):
     """Test values changed for climate device. Mapping."""
     device = device_mapping
-    assert device.hvac_mode == 'off'
-    device.values.mode.data = 'Heat'
-    value_changed(device.values.mode)
     assert device.hvac_mode == HVAC_MODE_HEAT
+    assert device.preset_mode == PRESET_NONE
+    device.values.mode.data = 'Off'
+    value_changed(device.values.mode)
+    assert device.hvac_mode == HVAC_MODE_OFF
+    assert device.preset_mode == PRESET_NONE
     device.values.mode.data = 'Cool'
     value_changed(device.values.mode)
     assert device.hvac_mode == HVAC_MODE_COOL
-    device.values.mode.data = 'Off'
+    assert device.preset_mode == PRESET_NONE
+
+
+def test_operation_value_changed_mapping_preset(device_mapping):
+    """Test values changed for climate device. Mapping with presets."""
+    device = device_mapping
+    assert device.hvac_mode == HVAC_MODE_HEAT
+    assert device.preset_mode == PRESET_NONE
+    device.values.mode.data = 'Heat Eco'
     value_changed(device.values.mode)
-    assert device.hvac_mode == HVAC_MODE_OFF
+    assert device.hvac_mode is None
+    assert device.preset_mode == PRESET_ECO
+    device.values.mode.data = 'Full Power'
+    value_changed(device.values.mode)
+    assert device.hvac_mode is None
+    assert device.preset_mode == PRESET_BOOST
+    device.values.mode = None
+    assert device.preset_mode == PRESET_NONE
+
```

Comment:
That is wrong. Eco heat need an hvac state of heat. Boost need set the correct state cool/heat

Reply:
Done

Reply:
If you think that is correct. You have the device and known what it showing correct on UI. My `cool/heat` was not point to `HEAT_COOLE` mode it was more like a PRESET_ECO and mode cool = "Eco Cool" and PRESET_ECO with mode heat = "Eco Heat"

Reply:
Thank you. I've explained my decision below